params.go

```diff
@@ -597,29 +624,56 @@ func (p *Params) ValidateRequiredProperties() (bool, []error) {
 		}
 	}
 
-	// validate sidecar params
-	if p.Sidecar.Type == "" {
+	// The "sidecar" field is deprecated, so it can be empty. But if it's specified, then we validate it.
+	if p.Sidecar.Type != "" && p.Sidecar.Type != "none" {
+		errors = validateSidecar(p.Sidecar, errors)
+	}
+
+	// validate sidecars params
+	for _, sidecar := range p.Sidecars {
+		errors = validateSidecar(sidecar, errors)
+	}
+
+	// Either the Sidecar has to be not empty, or Sidecars has to contain an element
+	if (p.Sidecar.Type == "" || p.Sidecar.Type == "none") && len(p.Sidecars) == 0 {
+		errors = append(errors, fmt.Errorf("Specifying a sidecar is required; Add one to the sidecars collection"))
+	}
+
+	return len(errors) == 0, errors
+}
+
+func validateSidecar(sidecar SidecarParams, errors []error) []error {
```

Comment:
Would prefer the following even though we don't use the Params object, it's more in line with the rest of the code.
```
func (p *Params) validateSidecar(sidecar SidecarParams, errors []error) []error
```